src/core.jl

```diff
@@ -427,6 +427,31 @@ function Base.map(f, As::AxisArray{T,N,D,Ax}...) where {T,N,D,Ax<:Tuple{Vararg{A
     return AxisArray(map(f, data...), As[1].axes...)
 end
 
+function Base.mapslices(f, A::AxisArray; dims)
+    new_axes = Axis[axes(A)...]
+
+    if (dims isa Integer || dims isa Vector{<:Integer})
+        for i in dims
+            ax = axes(A)[i]
+            new_axes[i] = ax(Base.OneTo(1))
+        end
+
+        return AxisArray(mapslices(f, A.data; dims=dims), new_axes...)
+    else
+        if (dims isa Axis || dims isa Type{<:Axis})
+            dims = [dims]
+        end
+
+        for ax in dims
+            i = axisdim(A, ax)
+            new_axes[i] = ax(Base.OneTo(1))
+        end
+
+        int_dims = axisdim.(Ref(A), dims)
+        return AxisArray(mapslices(f, A.data; dims=int_dims), new_axes...)
+    end
```

Comment:
Does this work? (if we also define `axisdim(A::AxisArray, i::Integer) = i`)
```suggestion
    dims =  dims isa Vector ? dims : [dims]
    for ax in dims
        i = axisdim(A, ax)
        new_axes[i] = ax(Base.OneTo(1))
    end

    int_dims = axisdim.(Ref(A), dims)
    return AxisArray(mapslices(f, A.data; dims=int_dims), new_axes...)
```

Reply:
yeah that's much neater - needed a slightly more robust method for `axisdims` in case `i` is out of bounds but it works a charm

Reply:
why do we need the error in that case? Can we just not leave it to hit the downstream error thrown by `mapslices`, which will probably be an informative `BoundsError`?

Reply:
if we can get away with it, I think not erroring would be consistent with behaviour in Base where e.g. `Base.size(rand(2,2), 5)` and `Base.axes(rand(2,2), 5)` don't error

Reply:
> Can we just not leave it to hit the downstream error thrown by mapslices, which will probably be an informative BoundsError?

We could, but in case anyone used that method in isolation they wouldn't necessarily hit any error.

>if we can get away with it, I think not erroring would be consistent with behaviour in Base where e.g. Base.size(rand(2,2), 5) and Base.axes(rand(2,2), 5) don't error

But if that's the case I'll remove it, although I'm surprised Base allows it.